lib/Sema/CSGen.cpp

```diff
@@ -3326,8 +3327,12 @@ namespace {
 
       // If the result type was declared IUO, add a disjunction for
       // bindings for the result of the coercion.
-      if (repr && repr->getKind() == TypeReprKind::ImplicitlyUnwrappedOptional)
-        return createTypeVariableAndDisjunctionForIUOCoercion(toType, locator);
+      if (repr &&
+          repr->getKind() == TypeReprKind::ImplicitlyUnwrappedOptional) {
+        auto *locatorIUO = CS.getConstraintLocator(expr);
```

Comment:
Nit: let's just inline this locator into the `createTypeVariableAndDisjunctionForIUOCoercion` call.